### ISSUE TITLE
Allowed html tags in default `Rich text` text format is insufficient

### ISSUE THREAD

**Issue by pandaskii** [govCMS/govcms8-installer#342]
Currently in GovCMS8

`
allowed_html: '<a href hreflang> <em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id class> <h3 id> <h4 id> <h5 id> <h6 id> <img src alt data-entity-type data-entity-uuid> <u> <s> <sup> <sub> <table> <caption> <tbody> <thead> <tfoot> <th> <td> <tr> <p> <drupal-entity alt title data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button>'
`

in GovCMS7, we have more HTML tags which are convenient for a new site

`
'valid_elements' => 'a[href|rel|target<_blank|title|class|name|id],
div[id|class|align<center?justify?left?right],
br,em,i,strong,cite,code,blockquote,ul,ol[class],li[class],dl,dt,dd,
span,p[class],address,
h2[class],h3[class],h4[class],h5[class],h6[class],
img[!src|title|alt|width|height|class|hspace|vspace],
table[border|cellpadding|cellspacing|align|dir|summary|style],tr[scope],thead,td[colspan|rowspan|headers|class|id],th[colspan|rowspan|class|headers|id|scope],tbody,caption,sup,sub,
@[class]
video',
`

**Comment by brett-sargeant** [comment]
This is editable by site admins in the config
` Home > Administration > Configuration > Content authoring `
then "configure" and scroll to the bottom of the page, look for "Limit allowed HTML tags and correct faulty HTML"

Note it won't allow you to use styles, you need to use classes for styling.

You will need to export the config for the change to be permanent.

This is the one I currently use (note I've removed italics)
`<a href hreflang data-entity-substitution data-entity-type data-entity-uuid id class> <b> <strong> <cite> <blockquote cite> <code> <ul type id class> <ol start type id class> <li id class> <dl id class> <dt id class> <dd id class> <h2 id class> <h3 id class> <h4 id class> <h5 id class> <h6 id class> <iframe width height src frameborder allowfullscreen allowFullScreen> <img id class src alt data-entity-type data-entity-uuid>  <s> <sup> <sub> <table id class> <caption id class> <tbody> <thead> <tfoot> <th id class> <td colspan rowspan id class> <tr id class> <p id class> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button> <div id class itemtype itemprop itemscope> <span id class title itemprop itemscope><nav id class><abbr title><colgroup><col span>`

**Comment by simesy** [comment]
```
<a href hreflang class target> <em> <strong> <cite> <blockquote cite> <code>
<ul type class id> <ol start type class id> <li> <dl> <dt> <dd> <h2 class id>
<h3 class id> <h4 class id>
<img class id src alt data-align data-caption data-entity-type data-entity-uuid data-responsive-image-style data-image-style width>
<u> <s> <sup> <sub> <table class id>
<caption> <tbody class id> <thead class id> <tfoot class id>
<th rowspan colspan class id> <td rowspan colspan class id> <tr class id>
<p class id>
<drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button>
<iframe src width height style name> <div class id>
```
Sharing is caring.

**Comment by simesy** [comment]
i would like to govcms being a bit more permissive by default. Maybe add a separate format for "expert" which is restricted roles and includes <iframe> and <script>

**Comment by simesy** [comment]
Ideally people like Brett and myself don't need to be porting around this list and applying to every new site - doesn't make any sense. :D

**Comment by brett-sargeant** [comment]
I think I saw a role for that in one of the sites handed over after the GovCMS financed migration.

edit: I don't mind applying it to our sites, as we include a lot of things that others don't need, as it makes the migration of our existing content easier due to our extensive use of classes.

**Comment by simesy** [comment]
Yeah but yours and mine are similar in many ways, so we can help new devs avoid the pain by pushing this into the distro. And when you have wysiwyg buttons not working because of text format - that is potentially painful and hard to debug.
